src/lib/evil/evil_libgen.c

```diff
@@ -0,0 +1,27 @@
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+#include <evil_private.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+
+EVIL_API char *
+evil_basename(char *path)
+{
+   // Expected from posix
+   if ((path == NULL) || (*path == '\0') ) return ".";
+
+   // Remove trailing '\\'
+   unsigned int len = strlen(path) -1;
+   if ((path[len] == '/') || (path[len] == '\\')) path[len] = '\0';
+
+   // Search for the last slash
+   while(--len) if ((path[len] == '/') || (path[len] == '\\')) break;
+
+   if(!len) return path;
+   else return &path[len + 1];
```

Comment:
What `path` is `"a"`? Wouldn't it get a simple `\0`?

Reply:
Thanks, fixed and added other tests.